### ISSUE TITLE
not validating a received transfer with an expiration date in the past

### ISSUE THREAD

**Issue by chinogarcia** [mojaloop/central-ledger#678]
**Summary**:
receives a transfer with an expiration date in the past, but the validation is not working

**Severity**:
Medium

**Priority**:
Medium

**Expected Behavior**
should validate that the expiration of a transfer is not in the past

**Steps to Reproduce**
  1. Send a quote 
  2. Send a transfer with the expiration 10 minutes ago
  3. the transfer reaches the payee simulator despite the transfer is expired

**Specifications**
  - Component (if known): central-ledger
  - Version: v8.7.1 / verified with OSS team version: v9.3.0
  - Platform: N/A
  - Subsystem: N/A
  - Type of testing: manual
  - Bug found/raised by: @chinogarcia

**Notes**:

- Severity when opened:
- Priority when opened:

**Comment by elnyry** [comment]
@chinogarcia - hi Damian, thanks for raising the issue; Can you please provide sample requests (code) for curl/postman requests used so that its easy to reproduce..

Also, please raise bugs here: https://github.com/mojaloop/project/issues/new/choose in future.

Coming to this issue here - the expiration date in context, is it the quote expiration (corresponding to that transfer you're referring to) or the transfer expiration in the **POST** /**transfers** request, thats not being validated here?

**Comment by chinogarcia** [comment]
hi @elnyry sure, its a simple quote and then a transfer but the expiration in the past.
For example:
```
curl --location --request POST 'mojaloop.workbench.mbox-dev-gke.io/quotes' \
--header 'Accept: application/vnd.interoperability.quotes+json' \
--header 'Content-Type: application/vnd.interoperability.quotes+json;version=1.0' \
--header 'Date: Mon, 13 Apr 2020 13:00:26 GMT' \
--header 'FSPIOP-Source: dfsp1' \
--header 'FSPIOP-Destination: dfsp4' \
--header 'FSPIOP-Signature: {"signature":"iU4GBXSfY8twZMj1zXX1CTe3LDO8Zvgui53icrriBxCUF_wltQmnjgWLWI4ZUEueVeOeTbDPBZazpBWYvBYpl5WJSUoXi14nVlangcsmu2vYkQUPmHtjOW-yb2ng6_aPfwd7oHLWrWzcsjTF-S4dW7GZRPHEbY_qCOhEwmmMOnE1FWF1OLvP0dM0r4y7FlnrZNhmuVIFhk_pMbEC44rtQmMFv4pm4EVGqmIm3eyXz0GkX8q_O1kGBoyIeV_P6RRcZ0nL6YUVMhPFSLJo6CIhL2zPm54Qdl2nVzDFWn_shVyV0Cl5vpcMJxJ--O_Zcbmpv6lxqDdygTC782Ob3CNMvg\",\"protectedHeader\":\"eyJhbGciOiJSUzI1NiIsIkZTUElPUC1VUkkiOiIvdHJhbnNmZXJzIiwiRlNQSU9QLUhUVFAtTWV0aG9kIjoiUE9TVCIsIkZTUElPUC1Tb3VyY2UiOiJPTUwiLCJGU1BJT1AtRGVzdGluYXRpb24iOiJNVE5Nb2JpbGVNb25leSIsIkRhdGUiOiIifQ"}' \
--header 'Authorization: Bearer xxx' \
--header 'Content-Type: text/plain' \
--data-raw '{
  "quoteId": "da5cc043-7b8c-4fd0-9480-2d88e828b48b",
  "transactionId": "d967a7e3-0a36-47c5-aa8b-746810c8c6b0",
  "payer": {
    "partyIdInfo": {
      "partyIdType": "MSISDN",
      "partyIdentifier": "1",
      "fspId": "dfsp1"
    },
    "personalInfo": {
      "complexName": {
        "firstName": "Mats",
        "lastName": "Hagman"
      },
      "dateOfBirth": "1983-10-25"
    }
  },
  "payee": {
    "partyIdInfo": {
      "partyIdType": "MSISDN",
      "partyIdentifier": "2",
      "fspId": "dfsp4"
    }
  },
  "amountType": "SEND",
  "amount": {
    "amount": "1",
    "currency": "EUR"
  },
  "transactionType": {
    "scenario": "TRANSFER",
    "initiator": "PAYER",
    "initiatorType": "CONSUMER"
  },
  "note": "hej"
}'
```

and then a transfer:
```
curl --location --request POST 'mojaloop.workbench.mbox-dev-gke.io/transfers' \
--header 'Accept: application/vnd.interoperability.transfers+json;version=1' \
--header 'Content-Type: application/vnd.interoperability.transfers+json;version=1.0' \
--header 'Date: Mon, 13 Apr 2020 13:00:45 GMT' \
--header 'FSPIOP-Source: dfsp1' \
--header 'FSPIOP-Destination: DFSPEUR' \
--header 'FSPIOP-Signature: {"signature":"iU4GBXSfY8twZMj1zXX1CTe3LDO8Zvgui53icrriBxCUF_wltQmnjgWLWI4ZUEueVeOeTbDPBZazpBWYvBYpl5WJSUoXi14nVlangcsmu2vYkQUPmHtjOW-yb2ng6_aPfwd7oHLWrWzcsjTF-S4dW7GZRPHEbY_qCOhEwmmMOnE1FWF1OLvP0dM0r4y7FlnrZNhmuVIFhk_pMbEC44rtQmMFv4pm4EVGqmIm3eyXz0GkX8q_O1kGBoyIeV_P6RRcZ0nL6YUVMhPFSLJo6CIhL2zPm54Qdl2nVzDFWn_shVyV0Cl5vpcMJxJ--O_Zcbmpv6lxqDdygTC782Ob3CNMvg\",\"protectedHeader\":\"eyJhbGciOiJSUzI1NiIsIkZTUElPUC1VUkkiOiIvdHJhbnNmZXJzIiwiRlNQSU9QLUhUVFAtTWV0aG9kIjoiUE9TVCIsIkZTUElPUC1Tb3VyY2UiOiJPTUwiLCJGU1BJT1AtRGVzdGluYXRpb24iOiJNVE5Nb2JpbGVNb25leSIsIkRhdGUiOiIifQ"}' \
--header 'FSPIOP-URI: /transfers' \
--header 'FSPIOP-HTTP-Method: POST' \
--header 'Content-Type: text/plain' \
--data-raw '{
  "transferId": "d967a7e3-0a36-47c5-aa8b-746810c8c6b0",
  "payerFsp": "dfsp1",
  "payeeFsp": "DFSPEUR",
  "amount": {
    "amount": "1",
    "currency": "EUR"
  },
  "expiration": "2020-04-13T12:50:45.818Z",
  "ilpPacket": "AYICwwAAAAAAAABkEGcuZGZzcDQubXNpc2RuLjKCAqZleUowY21GdWMyRmpkR2x2Ymtsa0lqb2laRGsyTjJFM1pUTXRNR0V6TmkwME4yTTFMV0ZoT0dJdE56UTJPREV3WXpoak5tSXdJaXdpY1hWdmRHVkpaQ0k2SW1SaE5XTmpNRFF6TFRkaU9HTXROR1prTUMwNU5EZ3dMVEprT0RobE9ESTRZalE0WWlJc0luQmhlV1ZsSWpwN0luQmhjblI1U1dSSmJtWnZJanA3SW5CaGNuUjVTV1JVZVhCbElqb2lUVk5KVTBST0lpd2ljR0Z5ZEhsSlpHVnVkR2xtYVdWeUlqb2lNaUlzSW1aemNFbGtJam9pWkdaemNEUWlmWDBzSW5CaGVXVnlJanA3SW5CaGNuUjVTV1JKYm1adklqcDdJbkJoY25SNVNXUlVlWEJsSWpvaVRWTkpVMFJPSWl3aWNHRnlkSGxKWkdWdWRHbG1hV1Z5SWpvaU1TSXNJbVp6Y0Vsa0lqb2laR1p6Y0RFaWZTd2ljR1Z5YzI5dVlXeEpibVp2SWpwN0ltTnZiWEJzWlhoT1lXMWxJanA3SW1acGNuTjBUbUZ0WlNJNklrMWhkSE1pTENKc1lYTjBUbUZ0WlNJNklraGhaMjFoYmlKOUxDSmtZWFJsVDJaQ2FYSjBhQ0k2SWpFNU9ETXRNVEF0TWpVaWZYMHNJbUZ0YjNWdWRDSTZleUpoYlc5MWJuUWlPaUl4SWl3aVkzVnljbVZ1WTNraU9pSkZWVklpZlN3aWRISmhibk5oWTNScGIyNVVlWEJsSWpwN0luTmpaVzVoY21sdklqb2lWRkpCVGxOR1JWSWlMQ0pwYm1sMGFXRjBiM0lpT2lKUVFWbEZVaUlzSW1sdWFYUnBZWFJ2Y2xSNWNHVWlPaUpEVDA1VFZVMUZVaUo5ZlEA",
  "condition": "Z_uqwVW6EkjSLvQp2fK3njM861f9ogyZ6erZ4yndSCc"
}'
```

As you can see the expiration in the body of the transfer is in the past

And the problem is with `POST /transfers`

**Comment by JoSowden** [comment]
Can we make this a high priority please

**Comment by elnyry** [comment]
@JoSowden sure, its now in progress

**Comment by elnyry** [comment]
@JoSowden @chinogarcia - should be available with central-ledger:v10.3.0 - the fix is in place and has been validated as reflected in issue: https://github.com/mojaloop/project/issues/1416
